Wikipedia/Code/WMFArticlePreviewTableViewCell.m

```diff
@@ -45,6 +60,28 @@ - (void)awakeFromNib {
     [self.saveButton setTitleColor:[UIColor wmf_blueTintColor] forState:UIControlStateNormal];
     self.saveButtonController.button = self.saveButton;
     [self wmf_setSelectedBackground];
+
+    [self setupBlurViewAndLoadingIndicator];
+    [self unblurAndHideLoadingIndicator];
+}
+
+- (void)setupBlurViewAndLoadingIndicator {
+    UIBlurEffect* blurEffect = [[UIBlurEffect alloc] init];
+    self.blurView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    [self.contentView addSubview:self.blurView];
+    [self.contentView bringSubviewToFront:self.blurView];
+    [self.blurView mas_makeConstraints:^(MASConstraintMaker* make) {
+        make.edges.equalTo(self.contentView);
+    }];
+
+    self.activityIndicator = [[UIActivityIndicatorView alloc] init];
```

Comment:
you don't have too… was just curious if you constrained the hight or width, or leading or trailing, etc… anything that would have stretched it.